### PR TITLE
Simplify install process to just use Composer

### PR DESCRIPTION
**Pull request by jonrandy**
#### 1. Objective

The previous install process was rather long winded and contained many unnecessary manual steps. Also, it was not the install method suggested/recommended by Magento

#### 2. Description of change

Modified `composer.json` with all necessary changes to allow full Composer-based installation (including dependencies). Also added keywords etc. to improve the discoverability of the plugin on Packagist

#### 3. Quality assurance

Before we go live with this, I would suggest at least a couple of people do an install from this `composer-install` branch. I have included the `composer.json` from my Magento project that I used to do this test install:
```
{
    "name": "magento/project-community-edition",
    "description": "eCommerce Platform for Growth (Community Edition)",
    "type": "project",
    "version": "2.2.2",
    "license": [
        "OSL-3.0",
        "AFL-3.0"
    ],
    "require": {
        "magento/product-community-edition": "2.2.2",
        "composer/composer": "@alpha",
        "omise/omise-magento": "dev-composer-install"
    },
    "require-dev": {
        "phpunit/phpunit": "~6.2.0",
        "squizlabs/php_codesniffer": "3.1.1",
        "phpmd/phpmd": "@stable",
        "pdepend/pdepend": "2.5.0",
        "friendsofphp/php-cs-fixer": "~2.2.0",
        "lusitanian/oauth": "~0.8.10",
        "sebastian/phpcpd": "2.0.4"
    },
    "autoload": {
        "psr-4": {
            "Magento\\Framework\\": "lib/internal/Magento/Framework/",
            "Magento\\Setup\\": "setup/src/Magento/Setup/",
            "Magento\\": "app/code/Magento/"
        },
        "psr-0": {
            "": [
                "app/code/"
            ]
        },
        "files": [
            "app/etc/NonComposerComponentRegistration.php"
        ],
        "exclude-from-classmap": [
            "**/dev/**",
            "**/update/**",
            "**/Test/**"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Magento\\Sniffs\\": "dev/tests/static/framework/Magento/Sniffs/",
            "Magento\\Tools\\": "dev/tools/Magento/Tools/",
            "Magento\\Tools\\Sanity\\": "dev/build/publication/sanity/Magento/Tools/Sanity/",
            "Magento\\TestFramework\\Inspection\\": "dev/tests/static/framework/Magento/TestFramework/Inspection/",
            "Magento\\TestFramework\\Utility\\": "dev/tests/static/framework/Magento/TestFramework/Utility/"
        }
    },
    "minimum-stability": "stable",
    "repositories": [
        {
            "type": "composer",
            "url": "https://repo.magento.com/"
        },
        {
            "type": "git",
            "url": "https://github.com/omise/omise-magento.git",
            "packagist.org": false
        }
    ],
    "extra": {
        "magento-force": "override"
    },
    "config": {
        "cache-dir": "/var/www/html/.composer/cache"
    }
}
```
The relevant things here for doing the install from the GitHub branch are:

1. Adding the ‘git’ item in the repositories section
2. Adding `"omise/omise-magento": "dev-composer-install"` to the `require` section. This tells composer to user the `composer-install` branch from the repo

#### 4. Impact of the change
Obviously the install instructions for the merchants now be different, so the https://www.omise.co/magento-2 page will need to be updated to reflect this

#### 5. Priority of change

Normal.

#### 6. Additional Notes

Black Panther was better than I thought it would be 😄

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+    "description": "Accept payments on your Magento 2 website with Omise",
+    "keywords": ["omise", "payment", "debitcard", "creditcard", "magento"],
+    "homepage": "https://github.com/omise/omise-magento",
+    "authors": [
+        {
+          "name": "Omise",
+          "email": "support@omise.co"
+        }
+    ],
+    "version": "2.4.0-dev",
```
Comment (guzzilar) [inline composer.json]: @jonrandy Omise-Magento still under `2.3.0-dev` version (we haven't released v2.3.0 yet), can you help change it back?

File: composer.json [hunk 2]
```diff
@@ -1,10 +1,22 @@
 {
     "name": "omise/omise-magento",
-    "version": "2.3.0-dev",
+    "description": "Accept payments on your Magento 2 website with Omise",
+    "keywords": ["omise", "payment", "debitcard", "creditcard", "magento"],
```
Comment (guzzilar) [inline composer.json]: @jonrandy nice for this! can you also add `internetbanking`? We also support that one.

File: composer.json [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
+    "homepage": "https://github.com/omise/omise-magento",
+    "authors": [
+        {
+          "name": "Omise",
+          "email": "support@omise.co"
+        }
+    ],
+    "version": "2.4.0-dev",
+    "minimum-stability": "stable",
```
Comment (guzzilar) [inline composer.json]: @jonrandy its default is already `stable`, do we still need this one?

File: composer.json (hunk 1 above)
Comment (jonrandy) [reply composer.json]: Aha, yeah - oops. I'd changed that when testing to make double sure Composer pulled my new version

File: composer.json (hunk 2 above)
Comment (jonrandy) [reply composer.json]: ok

File: composer.json (hunk 3 above)
Comment (jonrandy) [reply composer.json]: I was going for a "belt and braces" approach

**Comment by guzzilar** [comment]
I haven't try to pull it on my machine but look good 👍 (will test install after this)
cc @jacekstan

**Comment by jonrandy** [comment]
Merge? Bear in mind we need to update the Omise website as soon as we merge. Or do we? The old install instructions will still work I think

**Comment by guzzilar** [comment]
@jonrandy we are all remote today, maybe tomorrow?

**Comment by jonrandy** [comment]
@guzzilar Yep. I'll do it in office tomorrow then

**Pull request merged**